halo2_proofs/src/plonk/circuit.rs

```diff
@@ -470,14 +470,34 @@ pub trait Circuit<F: Field> {
     /// `Circuit` trait because its behaviour is circuit-critical.
     type FloorPlanner: FloorPlanner;
 
+    /// Runtime parameters to configure the circuit.
+    #[cfg(feature = "circuit-params")]
+    type Params: Default = ();
```

Comment:
Associated type default: https://github.com/rust-lang/rust/issues/29661

Reply:
@parazyd Would turning on this unstable feature - associated type default - be a good idea? 

I think tradeoff is:

the pro:
- won't break any circuits implementing the Circuit trait from zcash/halo2, which seems nice, one can use zcash/halo2 `Circuit` implementing circuits with this fork directly

the cons:
- potential corner cases (though the usage is simple here)
- need to migrate if and when Rust team introduces a breaking change (but the alternative is breaking the circuits now)

other misc points:
- Using an unstable feature requires depending on `nightly` which we already do in CI, so I think it isn't an issue.